src/frontend/src/binder/relation/mod.rs

```diff
@@ -137,6 +137,28 @@ impl Relation {
             _ => vec![],
         }
     }
+
+    pub fn visit_all_scan_table(&self, visitor: &mut impl FnMut(TableId)) {
+        match self {
+            Relation::Source(_) => {}
+            Relation::BaseTable(table) => {
+                visitor(table.table_id);
+            }
+            Relation::SystemTable(_) => {}
+            Relation::Subquery(subquery) => {
+                subquery.query.body.visit_all_scan_table_id(visitor);
+            }
+            Relation::Join(join) | Relation::Apply(join) => {
+                join.left.visit_all_scan_table(visitor);
+                join.right.visit_all_scan_table(visitor);
+            }
+            Relation::WindowTableFunction(_) => {}
+            Relation::TableFunction { .. } => {}
+            Relation::Watermark(_) => {}
+            Relation::Share(_) => {}
+            Relation::BackCteRef(_) => {}
```

Comment:
Why do we exclude these arms? IIUC, there can be relations defined in them.

Reply:
I have added logic to handle the rest arm. Didn't expect to have that many cases to handle.